### ISSUE TITLE
Failure to login

### ISSUE THREAD

**Issue by Jenna3715** [desktop/desktop#3125]
### Description

It fails to log me in.

### Version

**GitHub Desktop version:** v1.0.4 

**OS version:** Microsoft Windows [Version 10.0.15063]

### Steps to Reproduce

1. Open "Options" (File > Options...) (CTRL+,)
2. Under the "Accounts" tab, in the "GitHub.com" section, click on "Sign in"
3. Enter login info and click "Sign in"
4. Error box pops up saying "Failed to fetch"
5. Note: Signing in by clicking "Sign in using your browser" causes the login dialog box to display a banner that says "Unable to fetch authenticated user."

**Expected behavior:** It should log me in

**Actual behavior:** "Failed to fetch" 
**Reproduces how often:** 100%
### Logs

[2017-10-23.desktop.production.log](https://github.com/desktop/desktop/files/1405672/2017-10-23.desktop.production.log)

#### Additional Information

It would appear to me that the error is "git clone failed: exit status 128" or "error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128." Maybe both?

**Comment by shiftkey** [comment]
@Jenna3715 thanks for the information. This appears to be a network issue.

After the browser signin flow completes, we try and use the received token to fetch your account information - that seems to fail:

```
2017-10-23T03:00:15.384Z - info: [ui] [SignInStore] initializing OAuth flow
2017-10-23T03:00:19.108Z - info: [ui] [Dispatcher] requesting authenticated user
2017-10-23T03:00:19.357Z - warn: [ui] requestOAuthToken: failed with endpoint https://api.github.com
TypeError: Failed to fetch
2017-10-23T03:00:19.832Z - info: [ui] [SignInStore] error with OAuth flow
Error: Unable to fetch authenticated user.
    at t.Dispatcher.dispatchURLAction (C:\Users\jonah.sloan\AppData\Local\GitHubDesktop\app-1.0.4\webpack:\app\src\lib\dispatcher\dispatcher.ts:781:32)
    at <anonymous>
```

A clone also fails:

```
fatal: unable to access 'https://github.com/Jenna3715/ChatBot.git/': Failed to connect to github.com port 443: Bad access
```

Are you behind a proxy server of some sort?

**Comment by Jenna3715** [comment]
@shiftkey 
I just checked and my computer does not appear to be behind a proxy, but the WiFi I am using is supplied by a college campus. Whether that makes a difference, I wouldn't know.

**Comment by Jenna3715** [comment]
Flushing the caches and disabling the firewall on my computer didn't seem to help either.

**Comment by Jenna3715** [comment]
@shiftkey With the sign in flow, I tried both Firefox and Edge, but it failed with both. Also, my computer has Kaspersky Anti-Virus installed, and is working under a policy, so it's not very easy to turn it off.

**Comment by joshaber** [comment]
Whoops, looks like our bot got a little too aggressive. Re-opening.

**Comment by NegativeZero000** [comment]
For what this is worth I was also facing the issue reported by the OP. I also have Kaspersky Endpoint Security 10 for Windows. Once I temporarily disable the protection control in Kaspersky I can then authenticate successfully using the "Sign in using your browser" option. This also appears to be the reason I got stuck signing in directly with the client as per https://github.com/desktop/desktop/issues/2769. If I can figure out what exception I need to make I will report back.

**Comment by iAmWillShepherd** [comment]
@NegativeZero000

Thanks for the additional info. Do you by any chance have access to the logs from the day you experienced this issue?

**Comment by tmeeha** [comment]
Updated to the latest Github Desktop today, from an older version.  After install could not log in and clone repos via new version.  After 4 frustrating hours of googling and troubleshooting, I reinstalled an old version and was able to log in and clone repos no problem.  Just found this thread and tried turning off my Kaspari, reinstalling the new Github Desktop, and now I can log in and clone repos.  So, very similar problem as NegativeZero000.

**Comment by NegativeZero000** [comment]
@iAmWillShepherd I have attached my recent log and a screenshot of my latest attempt. 

[2017-12-05.desktop.production.log](https://github.com/desktop/desktop/files/1531030/2017-12-05.desktop.production.log)

[screenshot](https://user-images.githubusercontent.com/14927596/33606240-cf9ac578-d98a-11e7-91eb-9711bb3e2b9d.png)

Still no luck on figuring out which component of KES is causing the issue.

**Comment by shiftkey** [comment]
@NegativeZero000 that mention of port 443 is interesting, because it looks like checking for updates in the app also has a problem:

```
Error: Command failed: 4294967295
System.AggregateException: One or more errors occurred. ---> System.Net.WebException: Unable to connect to the remote server ---> System.Net.Sockets.SocketException: An attempt was made to access a socket in a way forbidden by its access permissions 52.2.53.204:443
```

Which is more descriptive than what we get from inside the Electron app itself:

```
2017-12-05T11:58:10.527Z - warn: [ui] requestOAuthToken: failed with endpoint https://api.github.com
TypeError: Failed to fetch
```

I guess this workaround is the best we can do currently:

> Kaspersky asked me to turn on monitoring on port 443. In doing so GitHub sign in process looks to be working fine. 

I'll leave this open for @iAmWillShepherd to see if there's anything else to do here before we close it out as a known issue.

**Comment by iAmWillShepherd** [comment]
Nothing to add here. I was unable to reproduce, likely due to my lack of anti-virus software in my Windows VM.

**Comment by jpgsantos** [comment]
Hello I have the same problem mentioned her but it only manifests itself when I try to login to an enterprise account, the non enterprise works fine, I've tried using chrome, Mozilla and edge but it always gives the same error.

I have no antivirus installed and both the built in antivirus and firewall from Microsoft are disabled.

GitHub desktop version 
1.0.13
Windows 10 version
16299.125
![image](https://user-images.githubusercontent.com/19508275/35625907-4b45adc4-0694-11e8-853a-31c6e8cb332a.png)

**Comment by iAmWillShepherd** [comment]
@jpgsantos could you open a new issue for your problem? After viewing your logs, your inability to login is different from this issue's.

**Comment by zzzgit** [comment]
i have the same problem 


[2018-04-19.desktop.production.log](https://github.com/desktop/desktop/files/1926789/2018-04-19.desktop.production.log)

**Comment by htran81** [comment]
![image](https://user-images.githubusercontent.com/22517191/48671543-60e5ed80-eb5c-11e8-9f1d-2679f897f8f4.png)
I'm facing the same problem while set up a new environtment.
Windows 10 - Chrome Browser - GitHub Desktop

**Comment by shiftkey** [comment]
@htran81 please open a new bug report and fill out the issue template - there are some details like error logs that will help us with troubleshooting

**Comment by gangadharanaik** [comment]
![image](https://user-images.githubusercontent.com/6031132/54806231-6a649b00-4c9f-11e9-8353-4044856ee826.png)

**Comment by steveward** [comment]
@gangadharanaik could you open up a new issue and fill out the template? Be sure to include the relevant log files as well, as that will provide us more information to debug this. Thanks!

**Comment by mishrabinash** [comment]
I am facing this issue only after clearing .AppData. Everything was fine before.

**Comment by shiftkey** [comment]
@mishrabinash please open a new [bug report](https://github.com/desktop/desktop/issues/new?template=bug_report.md) and fill out the template so that we're able to investigate the issue.

**Comment by Allencs16** [comment]
I have the same problem

**Comment by sshakil-wave** [comment]
Same here:

2021-01-18T19:45:25.512Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=...
2021-01-18T19:45:55.513Z - info: [main] Processing protocol url
2021-01-18T19:45:55.514Z - info: [ui] [Dispatcher] requesting authenticated user
2021-01-18T19:48:20.758Z - info: [ui] [AppStore] loading 0 repositories from store
2021-01-18T19:48:21.429Z - info: [ui] launching: 2.6.2 (Mac OS 10.16.0)
2021-01-18T19:48:21.429Z - info: [ui] execPath: '/Applications/GitHub Desktop.app/Contents/Frameworks/GitHub Desktop Helper (Renderer).app/Contents/MacOS/GitHub Desktop Helper (Renderer)'
2021-01-18T19:48:33.049Z - info: [ui] [SignInStore] initializing OAuth flow
2021-01-18T19:48:33.049Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=...
2021-01-18T19:48:35.026Z - info: [main] Processing protocol url
2021-01-18T19:48:35.039Z - info: [ui] [Dispatcher] requesting authenticated user
2021-01-18T19:49:05.073Z - warn: [ui] requestOAuthToken: failed with endpoint https://api.github.com
TypeError: Failed to fetch
2021-01-18T19:49:05.075Z - info: [ui] refocusing the main window after the OAuth flow is completed
2021-01-18T19:49:05.076Z - info: [ui] [SignInStore] error with OAuth flow
Error: Unable to fetch authenticated user.
    at Object.dispatchURLAction (/webpack:/app/src/ui/dispatcher/dispatcher.ts:1661:32)
2021-01-18T19:49:22.434Z - error: [ui] Error checking for updates
Error: The request timed out.

**Comment by ursula9** [comment]
same problem and I also have kaspersky security endpoint that i can turn off

**Comment by Fishish** [comment]
Turned off my Kaspersky and it worked.....

**Comment by HomemadeDev** [comment]
Temporarily disabling the two firewalls on my PC helped. Thanks. 
Try to check if your antivirus programs have firewalls too - it did in my case, and that's what caused my issue.
